internal/cli/global_opts.go

```diff
@@ -51,14 +52,20 @@ func (opts *GlobalOpts) ConfigOrgID() string {
 type cmdOpt func() error
 
 // PreRunE is a function to call before running the command,
-// this will validate the project ID and call any additional function pass as a callback
+// this will validate the projectID or orgID and call any additional function pass as a callback
 func (opts *GlobalOpts) PreRunE(cbs ...cmdOpt) error {
-	if opts.ConfigProjectID() == "" {
-		return errMissingProjectID
-	}
-	if err := validate.ObjectID(opts.ConfigProjectID()); err != nil {
-		return err
+	if opts.PreRunEOrg {
+		err := validateOrgID(opts)
+		if err != nil {
+			return err
+		}
+	} else {
+		err := validateProjectID(opts)
+		if err != nil {
+			return err
+		}
```

Comment:
I'd rather stay away from an `if` here, this avoids the question, why `opts.PreRunEOrg` and not `opts.PreRunEProj` or why not both?
I think I mentioned this in the ticket, we should aim for a full [functional config](https://sagikazarmark.hu/blog/functional-options-on-steroids/) here and pushing the need to validate orgs or projects when calling `PreRunE`, as `opts.PreRunE(opts.validateProjectID, ...)`

Reply:
yes, I was looking for a way to change less files as possible. Thanks for the feedback!

Reply:
I understand the fear and is one of the reasons we have not done this so far but the longer we wait the worst it'll get